src/app/dal/https-client.js

```diff
@@ -35,9 +40,9 @@ const httpsRequest = (
         });
         incomingMessage.on('end', () => {
           try {
-            if (validResponsePayload.statusCode === 204) {
-              resolve('');
-            } else if (isSuccessfulResponse(validResponsePayload.statusCode)) {
+            if (validResponsePayload.statusCode !== 204
+              && isSuccessfulResponse(validResponsePayload.statusCode)
+              && (!validResponsePayload.headers["Content-Type"] || validResponsePayload.headers["Content-Type"].indexOf("json") != -1)) {
```

Comment:
instead only parse if the status is success, the payload is json(or not defined) and the HTTP status code isn't no-content.